tooling/hibernate-enhance-maven-plugin/src/main/java/org/hibernate/orm/tooling/maven/MavenEnhancePlugin.java

```diff
@@ -168,6 +173,12 @@ public boolean doExtendedEnhancement(UnloadedClass classDescriptor) {
 			}
 			for ( File file : sourceSet ) {
 
+				String className = file.getName().replace(".class", "");
```

Comment:
I think this part is incorrect.  I think it needs to stip file paths up to the package name and convert filesystem path separators into `.`., right?

Reply:
No, `file.getName()` will return only the file name and not the complete path. So, just removing `.class` from it works.

Reply:
Not normal, but what about same-named classes in different packages?

Reply:
Yes, in that case both will be enhanced as per current implementation.
Updated it to use the fully qualified class name using your implementation of `determineClassName`